workflow.md

```diff
@@ -100,10 +123,91 @@ Please note that actual version of project is defined by project settings,
+`vX.Y.Z` already):
+    a) `cardano-sl-X.Y-rc` should be renamed to `cardano-sl-X.Y`, `cardano-sl-X.Y-mid` immediately spawned
+    b) `cardano-sl-X.Y-mid` is immediately spawned from the same commit.
+    c) `genesis.bin`, `constants-*.yaml` should be changed.
+    d) `vX.Y.Z` should be deployed on `csl-testnet` cluster.
+    e) tada.iohk.io, daedaluswallet.io should be updated.
+
+Rules for `-rc` branch are same as for release branch, specifically:
+* `-rc` branch must be merged into `master` after every change.
+* `master` is never merged into `-rc` branch.
+* `-rc` branch must be protected.
+
+### Interaction between cardano-sl and projects dependent on it
+
+Some projects (e. g. `cardano-sl-explorer`, `daedalus`) depend on
+`cardano-sl`. In order to facilitate interaction with `cardano-sl` we
+use special branch `-mid` mentioned above. Explorer is used as an
+example in the rest of this subsection, but rules apply to everything
+dependent on `cardano-sl`.
+
+* Explorer must depend only on version of `cardano-sl` from release
+  branch or `-mid` branch, not `-rc` branch. This rule is imposed
+  because only these branches are generally stable and
+  well-tested. `-rc` branch corresponds to a version which is not
+  properly tested yet, hence explorer should not depend on it.
+* Explorer should always be launched with the same version of
+  `cardano-sl` as the version it depends on. That's because different
+  versions of `cardano-sl` may be incompatible in some way and it may
+  lead to different (inconsistent) behavior of `cardano-node` and
+  `cardano-explorer`.
+* Version `X.Y.Z` of explorer (located in `sl-explorer-X.Y` branch)
+  can use only one of `X.Y.*` versions of `cardano-sl`. Note that
+  `Fix` versions may differ.
+* If explorer developer wants to change something in `cardano-sl` for
+  explorer's version `X.Y.Z` they should make a PR to `cardano-sl-X.Y`
+  branch, wait until it's merged and then use newer version in
+  explorer.
+* `master` branch of explorer should depend on the latest `-mid`
+  branch of `cardano-sl`.
+* If explorer developers wants to change something in `cardano-sl` for
+  explorer's `master` they should make a PR to `cardano-sl-X.Y-mid` of
+  `cardano-sl` (where `X.Y` is latest version of `cardano-sl`).
+* `cardano-sl-X.Y.` should be frequently merged into
+  `cardano-sl-X.Y-mid`.
+* `cardano-sl-X.Y-mid` should be frequently merged to `master`.
+
+**TODO** Frankly, I am not sure I properly understand all this `-mid` stuff.
+Treat it as a draft which should be refined.
+**TODO** this workflow as described now is bad, because
+`sl-explorer-0.5` can be tested (and, well, can exist) only after
+`cardano-sl-0.5` is released. It's probably bad. On the other hand, we
+don't want explorer guys to use unstable (untested) version.
+**TODO** according to workflow described above there should be only
+one `-mid` branch at any time. Is it correct? So it's something like
```

Comment:
> If it's a case that we don't need multiple `-mid` branches

@jagajaga says that it's not the case, so probably we can't.

As I wrote in my big comment (which probably nobody will read because it's too big 😜), I see only one case when it may be useful to have multiple `-mid` branches: when some project (e. g. explorer) doesn't support the latest version of `cardano-sl`. However, even in this case I don't see why it may be necessary to have `-mid` branch associated with some old version. If explorer team wants to change something in the old version, they can just make a PR to branch with old version, not `-mid` branch associated with this old version.

So I would like to have some clarification about why we need to have multiple `-mid` branches.